internal/grpc/services/storageprovider/storageprovider.go

```diff
@@ -600,10 +600,22 @@ func (s *service) Delete(ctx context.Context, req *provider.DeleteRequest) (*pro
 		}
 	}
 
-	err := s.storage.Delete(ctx, req.Ref)
+	md, err := s.storage.GetMD(ctx, req.Ref, []string{})
+	if err != nil {
+		return &provider.DeleteResponse{
+			Status: status.NewInternal(ctx, "can't stat resource to delete"),
+		}, nil
+	}
+
+	err = s.storage.Delete(ctx, req.Ref)
 
 	return &provider.DeleteResponse{
 		Status: status.NewStatusFromErrType(ctx, "delete", err),
+		Opaque: &typesv1beta1.Opaque{
+			Map: map[string]*typesv1beta1.OpaqueEntry{
+				"opaque_id": &typesv1beta1.OpaqueEntry{Decoder: "plain", Value: []byte(md.Id.OpaqueId)},
```

Comment:
I believe we need to stat because the reference could contain a path (relative reference) in which case the `ResourceId` would point to the parent resource, not the actual subject.